### ISSUE TITLE
remove 'install/cancel installation' button morphing

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#8716]
In 19.09 when in the admin interface you click 'install' on a repository it starts installation and the button is changing depending on the state (cloning/installing deps etc) but it also morphs into 'cancel' button (i.e. if you double click it it will both install and uninstall the repo) without even announcing it.

I suggest we separate install/cancel button and convey the state of installation somewhere next to the button, but not in the button name.

**Comment by ElectronicBlueberry** [comment]
I just looked into this. An issue on the current dev version is that the cancel option does not work at all. The uninstall request is sent, but the installation proceeds, and the tool is not uninstalled once it finishes.

Unless we want to change this behavior in the back-end, a fix could be to disable the cancel option and display the status on a disabled button during installation.

**Comment by guerler** [comment]
@ElectronicBlueberry thanks for looking into it. I am surprised that the cancellation does not work, I believe it used to and I am wondering if that is due to changes in the backend we have not adapted the ui to yet.

**Comment by guerler** [comment]
Looks like the default view turned out to be more confusing now: ![image](https://user-images.githubusercontent.com/2105447/184735189-dc0bc1ee-dce8-48a2-a7ed-5129e677a251.png). In current release, it is much easier to find the clickable buttons. 1) Maybe disabled buttons should just be hidden and 2) the status should only be shown during installation.

**Comment by mvdbeek** [comment]
I don't find this confusing, but if you want to work on this feel free to assign yourself.

**Comment by mvdbeek** [comment]
> Maybe 1) disabled buttons should just be hidden and 2) the status should only be shown during installation.

To clarify, 1 is exactly what caused the problem in that issue. The status is definitely helpful, but it could be placed elsewhere (we used to have a symbol for that)

**Comment by nsoranzo** [comment]
> Maybe 1) disabled buttons should just be hidden

Wouldn't this get us back to the original issue reported here (button morphing)?

> 2) the status should only be shown during installation.

I like to see the status shown, I'd actually prefer to have it on a separate column w.r.t. the buttons, that can be used to sort the rows by status.

**Comment by guerler** [comment]
@mvdbeek, yes I think it was much easier before to find the click targets:

<img width="819" alt="image" src="https://user-images.githubusercontent.com/2105447/184870029-735d99fe-e74c-412a-a3bc-259dfe6f8d90.png">

Now the number of UI elements has quadrupled without an increase in functionality, the additional elements are mostly redundant and only color distinguishes them. I mean it literally reads: "Installed, Install, Uninstall, Reset, Uninstalled, Install,  Uninstall, Reset...".

**Comment by ElectronicBlueberry** [comment]
I can give this another shot with the provided feedback, having the status in a separate column with only one button, so we can see what approach is preferable. The `Reset` button does communicate better what it is for, so it would still be an improvement over showing the status on the button.

**Comment by mvdbeek** [comment]
> . The morphing into the status can be avoided but the button could still morph from "Install" to "Uninstall" and back.

It really can't, when you do this you'll easily trigger a race of multiple install / uninstall cycles

**Comment by dannon** [comment]
@ElectronicBlueberry Thanks!  Maybe we can get by with a single button while addressing the feedback from the original issue "but it also morphs into 'cancel' button (i.e. if you double click it it will both install and uninstall the repo) without even announcing it." by adding a more obvious state change and/or an interaction locktout (i.e. -- button is disabled for 500ms after clicking or on state change).

**Comment by guerler** [comment]
Thanks @ElectronicBlueberry for looking again at it. It should be minor adjustments, but I feel it would reduce the complexity. I prepped a commit and this is how it looks:

<img width="861" alt="image" src="https://user-images.githubusercontent.com/2105447/184883095-ff9cfe02-37d1-48d3-9b4e-6180a94ff001.png">

If ppl are ok with it, we could take this route, I prepped a draft commit at feel free to use it or parts of it however you like: https://github.com/galaxyproject/galaxy/commit/725ed8dae1eba58b4e6492f827c51a2aa3aa1f8d

Thank you!
